### PR TITLE
[micromega] Remove Simplex option and dead code

### PR DESCRIPTION
**Pull request by fajb**
The option was set by default and deprecated since 8.13.

- [x] Added **changelog**.
- [x] updated **documentation**.

### CODE REVIEW

File: doc/changelog/07-vernac-commands-and-options/15690-micromega-deprecate-Simplex-option.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Removed:**
+  `Simplex` flag, that was deprecated in 8.13
```
Comment (silene) [inline doc/changelog/07-vernac-commands-and-options/15690-micromega-deprecate-Simplex-option.rst:2]: The documentation says 8.14. Is 8.13 correct here? Also, you could remind here that it was the default behavior anyway.

Comment (fajb) [reply doc/changelog/07-vernac-commands-and-options/15690-micromega-deprecate-Simplex-option.rst:2]: You are right, this is 8.14. I'll update.

**Comment by ppedrot** [comment]
@fajb you should probably squash the commits.

**Review by ppedrot — APPROVED** [review]

**Review by silene — APPROVED** [review]

File: doc/changelog/07-vernac-commands-and-options/15690-micromega-deprecate-Simplex-option.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- **Removed:**
+  `Simplex` flag, that was deprecated in 8.14.
+  :tacn:`lia` and :tac:`lra` will always use the simplex solver (that was already the default behaviour).
```
Comment (ppedrot) [inline doc/changelog/07-vernac-commands-and-options/15690-micromega-deprecate-Simplex-option.rst:3]: This `:tac:` is a `:tacn:`

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**